File: yh_turtle/scripts/turtle_clear.py
#!/usr/bin/python
#-*- coding: utf-8 -*-

import rospy
from std_srvs.srv import Empty

class TurtleClear:
    def __init__(self):
        self.client = rospy.ServiceProxy("clear", Empty)

    def call(self):
        try:
            self.client()
            rospy.loginfo("Clear")
        except rospy.ServiceException as e:
            rospy.loginfo("Failde : %s", e)

if __name__ == "__main__":
    rospy.init_node("turtle_clear")
    turtle_clear = TurtleClear()
    turtle_clear.call()

File: yh_turtle/src/turtle_clear.cpp
#include "ros/ros.h"
#include "std_srvs/Empty.h"

struct MyClient
{
    public:
    MyClient()
    {
        
        client = nh.serviceClient<std_srvs::Empty>("clear");
    }    
    void call()
    {
        if (client.call(srv))
        {
            ROS_INFO("Clear");
        }
        else
        {
            ROS_ERROR("Failed");
        }
    }
    private:
    ros::NodeHandle nh;
    ros::ServiceClient client;
    std_srvs::Empty srv;
};

int main(int argc, char** argv)
{
    ros::init(argc, argv, "turtle_clear");
    MyClient my_client = MyClient();
    my_client.call();
    
    return 0;
}

